scripts/list_provider_vms.py

```diff
@@ -0,0 +1,145 @@
+import argparse
+from tabulate import tabulate
+from multiprocessing import Process, Queue
+from multiprocessing.queues import Empty
+
+from utils.conf import cfme_data
+from utils.path import log_path
+from utils.providers import get_mgmt, list_provider_keys
+
+
+# Constant for report
+NULL = '--'
+
+
+def parse_cmd_line():
+    """
+    Specify and parse arguments
+    :return: args, kwargs, the usual
+    """
+    parser = argparse.ArgumentParser(argument_default=None)
+    parser.add_argument('--outfile',
+                        default=log_path.join('list_provider_vms.log').strpath,
+                        dest='outfile')
+    parser.add_argument('--tag',
+                        default=None,
+                        dest='tag',
+                        action='append',
+                        help='A provider tag to match a group of providers instead of all '
+                             'providers from cfme_data. Can be used multiple times')
+    parser.add_argument('provider',
+                        default=None,
+                        nargs='*',
+                        help='Provider keys, can be user multiple times. If none are given '
+                             'the script will use all providers from cfme_data or match tags')
+
+    args = parser.parse_args()
+    return args
+
+
+def process_tags(provider_keys, tags=[]):
```

Comment:
You should be careful using mutable type as a default value in a function argument. Consider to use `None` instead.